compat/publish.js

```diff
@@ -0,0 +1,82 @@
+// SPDX-FileCopyrightText: 2022 Andre 'Staltz' Medeiros <contact@staltz.com>
+//
+// SPDX-License-Identifier: LGPL-3.0-only
+
+const ssbKeys = require('ssb-keys')
+const validate = require('ssb-validate')
+const Ref = require('ssb-ref')
+const { onceWhen } = require('../utils')
+
+exports.init = function (sbot, config) {
+  const hmacKey = null
+
+  function guardAgainstDuplicateLogs(methodName) {
+    if (sbot.db2migrate && sbot.db2migrate.doesOldLogExist()) {
+      return new Error(
+        'ssb-db2: refusing to ' +
+          methodName +
+          ' because the old log still exists. ' +
+          'This is to protect your feed from forking ' +
+          'into an irrecoverable state.'
+      )
+    }
+  }
+
+  function encryptContent(keys, content) {
+    if (
+      sbot.box2 &&
+      sbot.box2.supportsBox2 &&
```

Comment:
this doesn't seemt to be defined?

Reply:
Yeah it's not, but I'll come back to this file later. I want the implementation to just use ssb.db.create